File: sk_processor/src/models/stream_individual.py
from faust import Record
from models.algorithm import Algorithm

class StreamIndividual(Record,serializer="json"):
    id: str
    import_str: str
    alg_str: str
    db_training_path: str
    db_test_path: str
    parameters_str: list = None

    def to_algorithm_model(self):
        return Algorithm(self.import_str,self.alg_str,self.parameters_str)

File: ggp/src/main/kotlin/models/StreamIndividual.kt
package models

import com.google.gson.annotations.SerializedName

data class StreamIndividual(
    var id: String,
    @SerializedName("import_str")
    var importStr: String,
    @SerializedName("alg_str")
    var algStr: String,
    @SerializedName("db_training_path")
    var dbTrainingPath: String,
    @SerializedName("db_test_path")
    var dbTestPath: String,
    @SerializedName("parameters_str")
    var parameters: List<String> = emptyList()
)